posydon/binary_evol/SN/step_SN.py

```diff
@@ -503,22 +503,56 @@ def collapse_star(self, star):
                 elif getattr(star, MODEL_NAME_SEL) is None:
                     # NOTE: this option is needed to do the collapse
                     # for stars evolved with the step_detached or 
-                    # steo_disrupted.
+                    # step_disrupted.
                     # allow to continue with the collapse with profile
                     # or core masses
                     warnings.warn(f'{MODEL_NAME_SEL}: The collapsed star '
                                      'was not interpolated! If use_profiles '
                                      'or use_core_masses is set to True, '
                                      'continue with the collapse.')                 
                 else:
+                    # store some properties of the star object
+                    # to be used for collapse verification
+                    pre_SN_c_core_mass = star.c_core_mass
+                    pre_SN_he_core_mass = star.he_core_mass
+                    pre_SN_mass = star.mass
+                    
                     MODEL_properties = getattr(star, MODEL_NAME_SEL)
                     for key, value in MODEL_properties.items():
                         setattr(star, key, value)
-                
+                    
+
                     for key in STARPROPERTIES:
                         if key not in ["state", "mass", "spin",
                                         "m_disk_accreted ", "m_disk_radiated"]:
                             setattr(star, key, None)
+                            
+                    # check if SN_type is not None
+                    print('state:',star.state)
+                    print('type:',star.SN_type)  
+                    # check if SN_type matches the predicted CO
+                    # ie WD is no SN
+                    if star.state == 'WD' and star.SN_type != "WD":
+                        star.SN_type = 'WD'
+                    elif (star.state == "NS" or star.state == "BH") and \
+                         (star.SN_type != 'ECSN' or star.SN_type != 'CCSN'):
+                        star.SN_type = self.check_SN_type(pre_SN_c_core_mass,
+                                                pre_SN_he_core_mass,
+                                                pre_SN_mass)[-1]
```

Comment:
It might be the case, that the new SN_type is "WD", which would still be incompatible. I guess, in this case we should modify the state/CO_type to become "WD", shouldn't we?

Reply:
Correct. Logic changed which recalculates the SN_type and CO_type. And checks again if they are different. If they are, we force them the same. However, do we want to throw an error instead and fail the binary?

Reply:
I'd go with a warning. (Maybe you mean it that way, but for me an error is a reason for the code to not continue anymore.)